zebra-chain/src/orchard/arbitrary.rs

```diff
@@ -41,8 +41,13 @@ impl Arbitrary for note::Nullifier {
     fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
         use halo2::arithmetic::FieldExt;
 
-        (any::<u64>())
-            .prop_map(|number| Self::from(pallas::Scalar::from_u64(number).to_bytes()))
+        (vec(any::<u8>(), 64))
+            .prop_map(|bytes| {
+                let mut b = [0u8; 64];
+                b.copy_from_slice(bytes.as_slice());
+                Self::try_from(pallas::Scalar::from_bytes_wide(&b).to_bytes())
```

Comment:
Optional: this vec to array conversion is a bit clearer and simpler
```suggestion
                let bytes = bytes.try_into().expect("array is the correct length");
                Self::try_from(pallas::Scalar::from_bytes_wide(&bytes).to_bytes())
```